snapcraft/internal/meta/package_repository.py

```diff
@@ -120,13 +129,15 @@ def __init__(
         self,
         *,
         architectures: Optional[List[str]] = None,
-        components: List[str],
+        components: Optional[List[str]] = None,
         formats: Optional[List[str]] = None,
         key_id: str,
         key_server: Optional[str] = None,
         name: Optional[str] = None,
-        suites: List[str],
+        path: Optional[str] = None,
+        suites: Optional[List[str]] = None,
         url: str,
+        apt_repo: Type[Ubuntu] = Ubuntu,
```

Comment:
Is there a reason for this?

Reply:
There was for dependency injection, but I have since refactored install bits out of meta, so this will be dropped when I push.